dev/source-release.sh

```diff
@@ -20,72 +20,137 @@
 
 set -e
 
-if [ -z "$1" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
-fi
-
-if [ -z "$2" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
+usage () {
+    echo "usage: $0 [-k <key_id>] [-r <git_remote>] <version_num> <rc_num>"
+    echo "  -k      Specify signing key. Defaults to \"GPG default key\""
+    echo "  -r      Specify Git remote name. Defaults to \"origin\""
```

Comment:
I think this should continue to default to `apache` so it doesn't break existing workflows. `origin` is almost certainly not correct for a release because the standing convention for `origin` is that it is you personal remote.

Reply:
I think the use of origin here implies that one had `git clone`d this repo directly?

If the existing workflow uses `apache`, I’d say we stick to that. Also, possibly adding a small section to the docs for “Contributing” that provides instructions on setting the upstream remote (and names it “apache”) might be beneficial.

Some projects have documentation around this step, such as I believe ORC? I know I’ve seen it somewhere. It likely seems obvious / trivial to those of us who have been contributing to open source for a long time, but I don’t think there would be any harm in adding a section to the docs for contributing. It could outline forking and setting remote, as well as running local check style (or linking to the docs for that)?

I’ll try to find the other larger Apache project that explicitly documents how to set the upstream project.

Reply:
I changed the default remote back to `apache`.